library/src/interactor_impl.cxx

```diff
@@ -603,6 +826,73 @@ interactor& interactor_impl::setDropFilesCallBack(
   return *this;
 }
 
+//----------------------------------------------------------------------------
+interactor& interactor_impl::addCommandCallback(
+  std::string_view action, std::function<bool(const std::vector<std::string>&)> callback)
+{
+  this->Internals->CommandCallbacks[std::string(action)] = callback;
```

Comment:
In the end you have a string in the map so why do you use a string_view?  

I think this would make more sense:
```cpp
interactor& interactor_impl::addCommandCallback(
  std::string action, std::function<bool(const std::vector<std::string>&)> callback)
{
  this->Internals->CommandCallbacks[std::move(action)] = callback;
  return *this;
}
```
or
```cpp
interactor& interactor_impl::addCommandCallback(
  std::string&& action, std::function<bool(const std::vector<std::string>&)> callback)
{
  this->Internals->CommandCallbacks[action] = callback;
  return *this;
}
```
but that works only with rvalues.

Reply:
> why do you use a string_view?

Because wherever I would previously use a `const std::string&`, I should now use a `std::string_view` instead. This is how I understand string_view, I know there is more to it but I think its works in the general case.

>  std::string action

One copy too many

>  std::string&& action

Only rvalues


Whats wrong with string_view ? They are _designed to replace_  string const ref .

Reply:
nevermind, you are right, this API should not use string_view I think.